pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+"""Tests for how well Pulp can deal with duplicate uploads.
+
+This module targets `Pulp #1406`_ and `Pulp Smash #81`_. The test procedure is
+as follows:
+
+1. Create a new feed-less repository.
+2. Upload content and import it into the repository. Assert the upload and
+   import was successful.
+3. Upload identical content and import it into the repository.
+
+The second upload should silently fail for all Pulp releases in the 2.x series.
+
+.. _Pulp #1406: https://pulp.plan.io/issues/1406
+.. _Pulp Smash #81: https://github.com/PulpQE/pulp-smash/issues/81
+"""
+from __future__ import unicode_literals
+
+from pulp_smash import api, utils
+from pulp_smash.compat import urljoin
+from pulp_smash.constants import (
+    CONTENT_UPLOAD_PATH,
+    PUPPET_MODULE_URL,
+    REPOSITORY_PATH,
+)
+from pulp_smash.tests.puppet.api_v2.utils import gen_repo
+
+
+def _upload_import_puppet_module(server_config, puppet_module, repo_href):
+    """Upload a puppet module and import it into a repository.
+
+    Return the JSON-decoded body of the call report received when importing the
+    Puppet module.
+    """
+    client = api.Client(server_config, api.json_handler)
+    malloc = client.post(CONTENT_UPLOAD_PATH)
```

Comment:
This is a reference to the C `malloc()` function. When called, `malloc()` allocates the requested amount of memory and returns a pointer to the beginning of that section of memory. The user must later `free()` that memory. A failure to do so results in a memory leak, and a duplicate attempt to free memory results in a segfault or some other critical error.

Pulp's handling of content uploads is reminiscent of `malloc()`. When one makes this HTTP call, the server returns an href (kinda like the C pointer) that must later be used to end the upload request. A failure to do so is a resource leak, and a duplicate attempt results (should result?) in an error.